classes/client/object_client.php

```diff
@@ -42,19 +42,57 @@ public function register_stream_wrapper() {
      *
      * @return bool.
      */
-    public function support_signed_urls() {
+    public function support_presigned_urls() {
         return false;
     }
 
     /**
      * Generates pre-signed URL to storage file from its hash.
      *
      * @param string $contenthash File content hash.
+     * @param array $headers request headers.
      *
      * @throws \coding_exception
      */
-    public function generate_signed_url($contenthash, $filename) {
+    public function generate_presigned_url($contenthash, $headers) {
         throw new \coding_exception("Pre-signed URLs not supported");
     }
 
+    /**
+     * Checks that Content-Disposition represented in headers.
+     *
+     * @param array $headers request headers.
+     *
+     * @return bool
+     */
+    public function content_disposition_exists_in_headers($headers) {
+        foreach ($headers as $header) {
+            $found = strpos($header, 'Content-Disposition');
+
+            if ($found !== false) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns Content-Disposition header from headers.
+     *
+     * @param array $headers request headers.
+     *
+     * @return string file name.
+     * @throws \coding_exception
+     */
+    public function get_content_disposition_header($headers) {
```

Comment:
Is there much value in having the two functions?
content_disposition_exists_in_headers vs get_content_disposition_header

It could just return an empty string if there are none, and it's a falsy value